crates/router/src/types/domain/user.rs

```diff
@@ -1031,3 +1077,36 @@ impl RoleName {
         self.0
     }
 }
+
+#[derive(serde::Serialize, serde::Deserialize, Debug)]
+pub struct RecoveryCodes(pub Vec<Secret<String>>);
+
+impl RecoveryCodes {
+    pub fn generate_new() -> Self {
+        let mut rand = rand::thread_rng();
```

Comment:
Nit: Naming.

You can call this `rng` so that it doesn't cause confusion with the `rand` crate itself. You can take it up in your next PR, not a problem.